pkg/monitoring/metrics/virt-controller/vmstats_collector_test.go

```diff
@@ -368,4 +370,144 @@ var _ = Describe("VM Stats Collector", func() {
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceStorage: size,
+						},
+					},
+					StorageClassName: &storageClassName,
+				},
+			}
+		}
+
+		It("should collect PVC size metrics correctly", func() {
+			volumeMode := k8sv1.PersistentVolumeFilesystem
+			pvc := createPVC("default", "test-vm-pvc", resource.MustParse("5Gi"), &volumeMode)
+			err := persistentVolumeClaimInformer.GetIndexer().Add(pvc)
+			Expect(err).ToNot(HaveOccurred())
+
+			vm := &k6tv1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "test-vm",
+				},
+				Spec: k6tv1.VirtualMachineSpec{
+					Template: &k6tv1.VirtualMachineInstanceTemplateSpec{
+						Spec: k6tv1.VirtualMachineInstanceSpec{
+							Volumes: []k6tv1.Volume{
+								{
+									Name: "rootdisk",
+									VolumeSource: k6tv1.VolumeSource{
+										PersistentVolumeClaim: &k6tv1.PersistentVolumeClaimVolumeSource{
+											PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+												ClaimName: "test-vm-pvc",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			results := CollectDiskAllocatedSize([]*k6tv1.VirtualMachine{vm})
+
+			Expect(results).ToNot(BeEmpty())
+			Expect(results[0].Metric.GetOpts().Name).To(Equal("kubevirt_vm_disk_allocated_size_bytes"))
+			Expect(results[0].Value).To(Equal(float64(5 * 1024 * 1024 * 1024)))
+			Expect(results[0].Labels).To(Equal([]string{"test-vm", "default", "test-vm-pvc", "Filesystem", "rootdisk"}))
+		})
+
+		It("should handle PVC with nil volume mode", func() {
+			pvc := createPVC("default", "test-vm-pvc-nil-mode", resource.MustParse("3Gi"), nil)
+			err := persistentVolumeClaimInformer.GetIndexer().Add(pvc)
+			Expect(err).ToNot(HaveOccurred())
+
+			vm := &k6tv1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "test-vm-nil-mode",
+				},
+				Spec: k6tv1.VirtualMachineSpec{
+					Template: &k6tv1.VirtualMachineInstanceTemplateSpec{
+						Spec: k6tv1.VirtualMachineInstanceSpec{
+							Volumes: []k6tv1.Volume{
+								{
+									Name: "rootdisk",
+									VolumeSource: k6tv1.VolumeSource{
+										PersistentVolumeClaim: &k6tv1.PersistentVolumeClaimVolumeSource{
+											PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+												ClaimName: "test-vm-pvc-nil-mode",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			results := CollectDiskAllocatedSize([]*k6tv1.VirtualMachine{vm})
+
+			Expect(results).ToNot(BeEmpty())
+			Expect(results[0].Metric.GetOpts().Name).To(Equal("kubevirt_vm_disk_allocated_size_bytes"))
+			Expect(results[0].Value).To(Equal(float64(3 * 1024 * 1024 * 1024)))
+			Expect(results[0].Labels).To(Equal([]string{"test-vm-nil-mode", "default", "test-vm-pvc-nil-mode", "null", "rootdisk"}))
+		})
+
+		It("should handle PVC with different namespace and label values", func() {
```

Comment:
I don't think this differ from `should collect PVC size metrics correctly`. Can be dropped